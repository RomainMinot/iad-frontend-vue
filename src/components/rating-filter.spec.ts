import { mount } from '@vue/test-utils';

import RatingFilter from './rating-filter.vue';

describe(`rating-filter`, () => {
  it(`renders correctly rating filter`, (context) => {
    const wrapper = mount(RatingFilter, {
      global: { plugins: [context.router] },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it(`emits update:model-value event when rating filter is updated`, (context) => {
    const wrapper = mount(RatingFilter, {
      global: { plugins: [context.router] },
    });
    const filterValue = 3;
    const select = wrapper.findComponent({ name: `VSelect` });
    select.vm.$emit(`update:model-value`, filterValue);
    expect(select.emitted(`update:model-value`)).toBeTruthy();
    expect(select.emitted(`update:model-value`)?.[0]).toEqual([filterValue]);
  });

  it(`emits update event when rating filter value is changed`, (context) => {
    const wrapper = mount(RatingFilter, {
      global: { plugins: [context.router] },
    });
    const filterValue = 4;
    wrapper.vm.$emit(`update`, filterValue);
    expect(wrapper.emitted().update).toBeTruthy();
    expect(wrapper.emitted().update[0]).toEqual([filterValue]);
  });

  it(`updateRating function emits a value`, async (context) => {
    const wrapper = mount(RatingFilter, {
      global: { plugins: [context.router] },
    });
    const vm = wrapper.vm as any;
    vm.updateRating(4);
    expect(wrapper.emitted(`update`)).toBeTruthy();
    expect(wrapper.emitted(`update`)?.[0]).toEqual([4]);
    vm.updateRating(2);
    expect(wrapper.emitted(`update`)?.[1]).toEqual([2]);
    vm.updateRating(0);
    expect(wrapper.emitted(`update`)?.[2]).toEqual([0]);
  });
});
