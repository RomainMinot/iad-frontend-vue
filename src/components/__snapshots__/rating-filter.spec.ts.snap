// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rating-filter > renders correctly rating filter 1`] = `
"<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-outlined\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <div class=\\"v-progress-linear__indeterminate\\">
        <div class=\\"v-progress-linear__indeterminate long\\"></div>
        <div class=\\"v-progress-linear__indeterminate short\\"></div>
      </div>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-text\\">
    <div class=\\"flex flex-wrap items-center justify-between gap-8\\"><strong>Filter the restaurants by ratings</strong>
      <div class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--readonly v-text-field v-select v-select--single\\">
        <!---->
        <div class=\\"v-input__control\\">
          <div class=\\"v-field v-field--appended v-field--center-affix v-field--no-label v-field--variant-solo v-theme--light v-locale--is-ltr\\" role=\\"textbox\\">
            <div class=\\"v-field__overlay\\"></div>
            <div class=\\"v-field__loader\\">
              <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                <!---->
                <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
                <div class=\\"v-progress-linear__indeterminate\\">
                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                </div>
                <!---->
              </div>
            </div>
            <!---->
            <div class=\\"v-field__field\\" data-no-activator=\\"\\">
              <!----><label class=\\"v-label v-field-label\\" for=\\"input-0\\">
                <!---->
                <!---->
              </label>
              <!---->
              <div class=\\"v-field__input\\" data-no-activator=\\"\\">
                <!---->
                <!--teleport start-->
                <!--teleport end--><input readonly=\\"\\" placeholder=\\"all restaurants\\" size=\\"1\\" type=\\"text\\" id=\\"input-0\\" aria-describedby=\\"input-0-messages\\">
              </div>
              <!---->
            </div>
            <!---->
            <div class=\\"v-field__append-inner\\">
              <!----><i class=\\"mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon\\" aria-hidden=\\"true\\"></i>
              <!---->
            </div>
            <div class=\\"v-field__outline\\">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;
